---
title: "7_2_Analysis"
author: "Sabrina Rasch"
date: "20.01.2023"
output:
  html_document:
    df_print: paged
---

# Setup

```{r message=FALSE, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#set working directory to Source file location
  setwd("C:/Users/sabri/OneDrive/Desktop/HS22/RNA-sequencing/Project/Scripts/")

#load packages
  library(tidyverse)
  library(rtracklayer)
```

# Load data

```{r}
#Load results
result_table <- read.csv(file = "../7_Summary/results_all.csv", header = TRUE)
annotated_table <- result_table[which(is.na(result_table$gene_name) == FALSE), ]
novel_table <- result_table[which(is.na(result_table$gene_name)), ]


#Specify parameters for filtering
  width_threshold <- 200
  log2_fold_threshold <- 2
  q_val_threshold <- 0.025 # Because we are testing 2-sided! This mean the significance level is 0.05 
  prot_coding_pot_threshold <- 0.364
  total <- nrow(result_table)
  total_novel <- nrow(novel_table)
```

# Analysis

```{r}
#Check that there are only TRUE or FALSE for TSS, polyA and intergenic
  unique(result_table$TSS)
  unique(result_table$PolyA)
  unique(result_table$Intergenic)

#Proportion novel and annotated
  novel <- nrow(novel_table)
    prop_novel <- novel / total
  annotated <- nrow(annotated_table)
    prop_annotated <- annotated / total

#Proportion single-exon
  single_exon <- sum(result_table$Num_Exons == 1)
    prop_single_exon <- single_exon / total
  single_exon_novel <- sum(novel_table$Num_Exons == 1)
    prop_single_exon_novel <- single_exon_novel / total_novel

#Proportion coding vs. noncoding
  non_coding <- sum(result_table$prot_coding_pot <= prot_coding_pot_threshold)
    prop_non_coding <- non_coding / total
  non_coding_novel <- sum(novel_table$prot_coding_pot <= prot_coding_pot_threshold)
    prop_non_coding_novel <- non_coding_novel / total_novel

#Proportion TSS
  TSS <- sum(result_table$TSS)
    prop_TSS <- TSS / total
  TSS_novel <- sum(novel_table$TSS)
    prop_TSS_novel <- TSS_novel / total_novel

#Proportion polyA
  polyA <- sum(result_table$PolyA)
    prop_polyA <- polyA / total
  polyA_novel <- sum(novel_table$PolyA)
    prop_polyA_novel <- polyA_novel / total_novel

#Proportion "intergenic"
  intergenic <- sum(result_table$Intergenic)
    prop_intergenic <- intergenic / total
  intergenic_novel <- sum(novel_table$Intergenic)
    prop_intergenic_novel <- intergenic_novel / total_novel
    
#Proportion of long (>= 200nt) transcripts
  long <- sum(result_table$width >= width_threshold)
    prop_long <- long / total
  long_novel <- sum(novel_table$width >= width_threshold)
    prop_long_novel <- long_novel / total_novel
```

# Summarise and save

```{r}
#Create one result table
  result_stats <- c(total, single_exon, non_coding, TSS, polyA, intergenic, long)
  novel_stats <- c(total_novel, single_exon_novel, non_coding_novel, TSS_novel, polyA_novel, intergenic_novel, long_novel)
  annotated_stats <- result_stats - novel_stats

  result_table_stats <- as.data.frame(rbind(result_stats, novel_stats, annotated_stats))
    names(result_table_stats) <- c("total", "single_exon", "non_coding", "TSS", "polyA", "intergenic", "long")
      # write.csv(result_table_stats, "../7_Summary/result_stats.csv", row.names = FALSE)
      result_table_stats <- read.csv("../7_Summary/result_stats.csv")
```

# Proportions

```{r}
total <- result_table_stats[1, 1]
total_novel <- result_table_stats[2, 1]
total_annotated <- result_table_stats[3, 1]
  perc_novel <- (100 / total) * total_novel
  perc_annotated <- (100 / total) * total_annotated

result_table_stats[2, ] <- (100 / total_novel) * result_table_stats[2, ]
result_table_stats[3, ] <- (100 / total_annotated) * result_table_stats[3, ]
result_table_stats[1, ] <- (100 / total) * result_table_stats[1, ]
result_table_stats <- round(result_table_stats, digits = 3)
write.csv(result_table_stats, "../7_Summary/result_stats_perc.csv", row.names = FALSE)
```



